### ISSUE TITLE
Improve error message for Enum with no arguments

### ISSUE THREAD

**Issue by mdickinson** [enthought/traits#934]
When creating an `Enum` with no arguments, we get a non-obvious failure message:

```
>>> class A(HasTraits):
...     foo = Enum()
... 
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "<stdin>", line 2, in A
  File "/Users/mdickinson/Enthought/ETS/traits/traits/trait_types.py", line 1967, in __init__
    default_value = args[0]
IndexError: tuple index out of range
```

It would be good to fix the exception type and exception message to something better.

**Comment by mdickinson** [comment]
@midhun-pm Do you think it would make sense to try to solve this as part of #889, or separately?

**Comment by midhun-pm** [comment]
Fixed in PR #889

**Comment by mdickinson** [comment]
This was solved in #889 (thank you, @midhun-pm!)
